src/views/darkroom.c

```diff
@@ -947,18 +947,15 @@ static void _dev_change_image(dt_develop_t *dev, const int32_t imgid)
   // commit image ops to db
   dt_dev_write_history(dev);
 
+  const int32_t devid = dev->image_storage.id;
+
   // be sure light table will update the thumbnail
-  if(!dt_history_hash_is_mipmap_synced(dev->image_storage.id))
+  if(!dt_history_hash_is_mipmap_synced(devid))
   {
-    const dt_history_hash_t hash_status = dt_history_hash_get_status(dev->image_storage.id);
-
-    dt_mipmap_cache_remove(darktable.mipmap_cache, dev->image_storage.id);
-    dt_image_update_final_size(dev->image_storage.id);
-    const gboolean fresh = (hash_status == DT_HISTORY_HASH_BASIC) || (hash_status == DT_HISTORY_HASH_AUTO);
-    const dt_imageio_write_xmp_t xmp_mode = dt_image_get_xmp_mode();
-    if((xmp_mode == DT_WRITE_XMP_ALWAYS) || ((xmp_mode == DT_WRITE_XMP_LAZY) && !fresh))
```

Comment:
I cannot see where this complex check using DT_WRITE_XMP_ALWAYS, DT_WRITE_XMP_LAZY, DT_HISTORY_HASH_BASIC & DT_HISTORY_HASH_AUTO is now done. Can you give me some hint for the review? TIA.

Reply:
It's all in done in `dt_image_write_sidecar_file(const int32_t imgid)`

Basically we trust all image developing to set the `DT_HISTORY_HASH_CURRENT` bit flag so `dt_image_altered()` returns a TRUE. From my understanding of the codebase this is the intended functionality of that function (also my tests here did not show problems) so we can avoid the "complicated" tests.